go/arrow/array/table.go

```diff
@@ -126,3 +195,61 @@ func (a *Chunked) NewSlice(i, j int64) *Chunked {
 
 	return NewChunked(a.dtype, chunks)
 }
+
+// simpleTable is a basic, non-lazy in-memory table.
+type simpleTable struct {
+	rows int
+	cols []Column
+
+	schema *arrow.Schema
+}
+
+// NewTable returns a new basic, non-lazy in-memory table.
+// If rows is negative, the number of rows will be inferred from the height
+// of the columns.
+//
+// NewTable panics if the columns and schema are inconsistent.
+// NewTable panics if rows is larger than the height of the columns.
+func NewTable(schema *arrow.Schema, cols []Column, rows int) *simpleTable {
+	tbl := simpleTable{
+		rows:   rows,
+		cols:   cols,
+		schema: schema,
+	}
+	if tbl.rows < 0 {
+		switch len(tbl.cols) {
+		case 0:
+			tbl.rows = 0
+		default:
+			tbl.rows = tbl.cols[0].Len()
+		}
+	}
+
+	tbl.validate()
+
+	return &tbl
+}
+
+func (tbl *simpleTable) Schema() *arrow.Schema { return tbl.schema }
+func (tbl *simpleTable) NumRows() int          { return tbl.rows }
+func (tbl *simpleTable) NumCols() int          { return len(tbl.cols) }
+func (tbl *simpleTable) Column(i int) *Column  { return &tbl.cols[i] }
+
```

Comment:
Similar to the `Chunked` implementation, `Table` should have a `Retain` and `Release` methods

```go
// Retain increases the reference count by 1.
// Retain may be called simultaneously from multiple goroutines.
func (tbl *simpleTable) Retain() {
	atomic.AddInt64(&tbl.refCount, 1)
}

// Release decreases the reference count by 1.
// When the reference count goes to zero, the memory is freed.
// Release may be called simultaneously from multiple goroutines.
func (a *Chunked) Release() {
	debug.Assert(atomic.LoadInt64(&tbl.refCount) > 0, "too many releases")

	if atomic.AddInt64(&tbl.refCount, -1) == 0 {
		for i := range a.cols {
			cols[i].Release()
		}
		a.chunks = nil
		a.length = 0
		a.nulls = 0
	}
}
```

Reply:
done.